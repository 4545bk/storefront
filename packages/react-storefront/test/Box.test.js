/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'
import Box, { Vbox, Hbox } from '../src/Box'

describe('Box', () => {

  it('renders with no props', () => {
    const component = (
      <Box/>
    )

    expect(create(component).toJSON()).toMatchSnapshot()
  })
  
  it('fowards props to style', () => {
    const component = (
      <Box 
        height="100%" 
        width="100%"
      />
    )

    expect(create(component).toJSON()).toMatchSnapshot()
  })


  it('accepts css', () => {
    const component = (
      <Box 
        className="myClass" 
        classes={{ root: 'rootClass' }} 
      />
    )

    expect(create(component).toJSON()).toMatchSnapshot()
  })
})

describe('Vbox', () => {
  it('renders', () => {
    expect(create(<Vbox/>).toJSON()).toMatchSnapshot()
  })
})

describe('Hbox', () => {
  it('renders', () => {
    expect(create(<Hbox/>).toJSON()).toMatchSnapshot()
  })
})