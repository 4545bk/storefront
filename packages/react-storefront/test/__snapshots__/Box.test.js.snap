// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box accepts css 1`] = `
<div
  className="RSFBox-root-1 rootClass myClass"
  style={Object {}}
/>
`;

exports[`Box fowards props to style 1`] = `
<div
  className="RSFBox-root-1"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;

exports[`Box renders with no props 1`] = `
<div
  className="RSFBox-root-1"
  style={Object {}}
/>
`;

exports[`Hbox renders 1`] = `
<div
  className="RSFBox-root-1"
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
/>
`;

exports[`Vbox renders 1`] = `
<div
  className="RSFBox-root-1"
  style={
    Object {
      "flexDirection": "column",
    }
  }
/>
`;
